src/Twig/Extension/AuthRuntime.php

```diff
@@ -191,6 +191,54 @@ public function getProviders($guid = null)
     }
 
     /**
+     * Return an array of registered OAuth providers for an account with
+     * lst steen and last updated dates
+     *
+     * @param string $guid
+     *
+     * @return array
+     */
+    public function getProvidersDates($guid = null)
+    {
+        $providers = [];
+        if ($guid === null) {
+            $auth = $this->session->getAuthorisation();
+
+            if ($auth === null) {
+              return $providers;
+            }
+            $guid = $auth->getGuid();
+        }
+
+        $providerEntities = $this->records->getProvisionsByGuid($guid);
+        if ($providerEntities === false) {
+            return $providers;
+        }
+
+        /** @var Storage\Entity\Provider $providerEntity */
+        $providers['lastseen'] = null;
+        $providers['lastupdate'] = null;
```

Comment:
Wouldn't this lead to providers being a mix between the name/lastseen/lastupdate for all providers and the lastseen/lastupdate for the entire user? Seems like it'd be better to keep them separated, and for this feature/fix we only need the last seen for a user, right?

Especially since the array would inconsistent in what types it contains and how the keys are structured.